src/FeatureDescriptors.jl

```diff
@@ -1,5 +1,52 @@
 module FeatureDescriptors
 
-# Write your package code here.
+export Descriptor
+export label, tables, value_col
+
+"""
+    Descriptor
+
+The `Descriptor`s interface lets users define the properties and relationships between the
+features being used in their models or other data-intensive workflows.
+
+`Descriptor` supertypes define some general properties that can be overloaded for specific
+subtypes as required, such as the tables to load or the value column to parse.
+
+Users may define their own Descriptor` type system for their own purposes but must extend
+the following methods:
+- [`tables`](@ref): The tables that contain the data needed to build the feature.
+- [`value_col`](@ref): The column name that contains the quantity of interest.
+"""
+abstract type Descriptor end
+
+"""
+    tables(::Descriptor) -> Vector{String}
+
+The name(s) of the table(s) associated with the [`Descriptor`](@ref), e.g.
+`["temperature.csv", "locations.csv"]`.
+"""
+function tables end
+
+"""
+    value_col(::Descriptor) -> Symbol
+
+The column containing the quantity of interest for the [`Descriptor`](@ref), e.g. `:temperature`.
+"""
+function value_col end
+
+"""
+    label(D::Descriptor) -> Symbol
+
+Return a symbol representation of the descriptor.
+"""
+label(D::Type{<:Descriptor}) = Symbol(_tostring(D))
+
+function _tostring(D::Type{<:Descriptor})
+    # Remove module for unexported Descriptors
+    s = replace(string(D), string(parentmodule(D)) => "")
+    return replace(lowercase(s), r"[^\w]" => "")
```

Comment:
Could comment this. It's removing any non-word (basically weird) characters right?